random1on1/api/channels.py

```diff
@@ -0,0 +1,28 @@
+import abc
+import discord
+
+
+class ReadWriteChannel(discord.channel.TextChannel, metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __init__(self, channel_name: str):
+        # TODO: Add precondition checks here
+        self.channel_name = channel_name
+
+
+class WriteOnlyChannel(discord.channel.TextChannel, metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __init__(self, channel_name: str):
+        # TODO: Add preconditions here
+        self.channel_name = channel_name
```

Comment:
Consider avoiding the use of classes if there are too few methods.

Reply:
Ah there's going to be significant differences in the permissions checking parts for these -- just didn't write them yet